main/tests/cypress/cypress/support/commands.js

```diff
@@ -239,10 +239,10 @@ Cypress.Commands.add('assertGridEquals', (values) => {
     const cells = Cypress.$('table.data-table tbody tr')
       .map(function (i, el) {
         return [
-          Cypress.$('td', el)
-            .filter((index) => index > 2)
+          Cypress.$('td', el).filter(index => index > 2)
             .map(function (index, element) {
-              return element.innerText;
+              const innerText = element.querySelector('div.data-table-cell-content div > span').innerText;
+              return innerText; // nulled fields in the grid has its text node set to "null" even through they are set to display: none.
```

Comment:
I think we should go with `'null'` because it's more true to what's being tested.

Reply:
I am confused - why? If a cell is `null`, then I would like tests to assert that the cell is `null`, not that it contains the string `"null"`, no?

I would say exactly the opposite as you: "we should go with `null` because it's more true to what's being tested"… so because I normally agree with you on things like this I suspect there is a misunderstanding somewhere!

Reply:
We aren't testing the state of the cell we are testing the DOM contents, by using `null` we are implying that the cell state is being tested which it is not.

Reply:
I see… I don't think I agree that we are testing the DOM state only. This is an end-to-end suite, so we are testing the operation logic as well. For me, it makes sense if those Cypress are written in a way which really reflects the experience of a user, and can be read as such. For a user, whether a cell contains `null` or `"null"` makes a really big difference, so I would like us not to conflate those two cases.

Conflating `null` and `""` (which seems to be what we have been doing until now in this helper) is more acceptable because that corresponds to the notion of "blank cells" which we use in the tool and therefore makes sense from a user perspective.

Reply:
I'm happy to change it but the behavior you describes is exactly what gave me a hard time figuring out what the test was actually testing going into this.

Reply:
No I think we would need to represent each cell with a custom object if we wanted to capture the cell state(which would actually be neat as we could make better tests for all the various cell states).

We could probably move "null" into an aria-label attribute, I will give it a go and if I find that it's out of scope we can just default to letting `'null'` be `null`.